src/puppetlabs/trapperkeeper/core.clj

```diff
@@ -142,10 +153,12 @@
   [cli-data]
   {:pre [(map? cli-data)]}
   (let [app (boot-with-cli-data cli-data)]
-    ;; This line populates the `main-app` variable with the TrapperkeeperApp
-    ;; instance.  This allows it to be referenced in a remote nREPL session.
-    (alter-var-root #'main-app (fn [_] app))
-    (run-app app)))
+    ;; This adds the TrapperkeeperApp instance to the tk-apps list, so that
+    ;; it can be referenced in a remote nREPL session, etc.
+    (swap! internal/tk-apps conj app)
+    (internal/register-sighup-handler)
```

Comment:
One thing I notice here:

We don't do the initial registration of the sighup handler until after the app is booted.  I think this lessens the concerns that people had about a HUP causing indeterminate behavior if the user sent it before the app had finished booting up.  The behavior won't be indeterminate -- it will just shut down the process.

We might want to revisit that in the future, and change this up so that the registration of the handler happens earlier *and* it's smart enough not to initiate the restart while the initial startup is in progress, but I think we can defer that for now.  The fact that the behavior is not indeterminate seems to alleviate the most pressing concerns.